changelog/unreleased/bugfixes/7464.md

```diff
@@ -0,0 +1 @@
+- Fixed `SetRoutesReason` when alternative route is chosen as a primary.
```

Comment:
I don't think we need a changelog entry here. This change does not affect the user at all. It's an internal change. But with the current phrasing they might think that the reason in RoutesUpdatedResult is now set differently.